jenkins/css-pipeline

```diff
@@ -167,14 +168,20 @@ pipeline {
             }
             ]
             }"""
-            server.upload(uploadSpec)
+            server.upload(uploadSpec, buildInfo)
             slackSend channel: "#cs-studio", message: "CS-Studio ver. ${versionNBR} loaded into Artifactory at ${artifactoryFolder}."
           }
         }
       }
-
-
-
+      stage('00900\ndeploy build information and retent old builds') {
+        when {
+          expression { env.repoBranch != 'production' }
+        }
+        steps {
+          buildInfo.retention maxBuilds: 7
```

Comment:
How does this affect production artifacts? Will this retain the artifacts linked to the 7 last builds in Artifactory AND any artifacts that are not linked to any builds?
What would happen if we wanted to add buildinfo for production-builds as well?

Reply:
Good questions, and I haven't considered this at all. But since it's two different jobs in Jenkins I would guess production builds wouldn't be affected by this. I'm not sure how to test it without risking to break anything though.

Reply:
* I would certainly guess that the retention policy is only active on builds, not on individual artifacts. So old artifacts should be safe.
* And since it's different builds in Jenkins, it should easily separate the builds well enough to only handle the development build artifacts and leave production builds alone.